packages/core/strapi/src/node/create-build-context.ts

```diff
@@ -79,15 +80,25 @@ const createBuildContext = async <TOptions extends BaseOptions>({
     });
 
   const serverUrl = strapiInstance.config.get<string>('server.url');
-  const adminPath = strapiInstance.config.get<string>('admin.path');
+  const serverAbsoluteUrl = strapiInstance.config.get<string>('server.absoluteUrl');
+  const adminUrl = strapiInstance.config.get<string>('admin.url');
+  const adminAbsoluteUrl = strapiInstance.config.get<string>('admin.absoluteUrl');
+
+  const sameOrigin = new URL(adminAbsoluteUrl).origin === new URL(serverAbsoluteUrl).origin;
+
+  const adminPublicPath = new URL(adminAbsoluteUrl).pathname;
+
+  const adminPath = sameOrigin
+    ? strings.getCommonPath(serverUrl, adminUrl)
+    : new URL(adminUrl).pathname;
 
   const appDir = strapiInstance.dirs.app.root;
 
   await loadEnv(cwd);
 
   const env = getStrapiAdminEnvVars({
-    ADMIN_PATH: adminPath,
-    STRAPI_ADMIN_BACKEND_URL: serverUrl,
+    ADMIN_PATH: adminPublicPath,
```

Comment:
Just to be sure, why are we using adminPublicPath instead of the computed adminPath here ?

Reply:
(might be worth adding a comment to explain the reasoning anyway)

Reply:
it's going to be the url used in on the browser side so you need to have the full path including the potential proxied parts